src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -1544,7 +1518,23 @@ public async IAsyncEnumerable<SourceFile> Load(SessionId id, string[] loaded_fil
                 }
                 catch (Exception e)
                 {
-                    logger.LogError($"Failed to load {step.Url} ({e.Message})");
+                    try
+                    {
+                        if (tryUseDebuggerProtocol)
+                        {
+                            string unescapedFileName = Uri.UnescapeDataString(step.Url);
+                            bytes = await context.SdbAgent.GetBytesFromAssemblyAndPdb(Path.GetFileName(unescapedFileName), token);
+                            assembly = new AssemblyInfo(monoProxy, id, step.Url, bytes[0], bytes[1], logger, token);
+                        }
+                        else
+                        {
+                            logger.LogDebug($"Failed to read {step.Url} ({e.Message})");
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        logger.LogError($"Failed to load {step.Url} ({ex.Message})");
```

Comment:
Wondering why `ex` is not passed as an argument to `LogError`? This might provide the user with more information?